java/main/src/main/java/com/saucelabs/saucebindings/VisualResults.java

```diff
@@ -0,0 +1,53 @@
+package com.saucelabs.saucebindings;
+
+import lombok.Getter;
+import lombok.Setter;
+import org.openqa.selenium.remote.RemoteWebDriver;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Setter @Getter
+public class VisualResults {
+    Map<String, Object> results;
+    private Map<String, Object> totals;
+
+    private final Boolean passed;
+    private final String message;
+    private final String status;
+    private final List<VisualSnapshot> snapshots = new ArrayList<>();
+    private final Long totalNew;
+    private final Long totalChanged;
+    private final Long totalRejected;
+    private final Long totalAccepted;
+    private final Long total;
+
+    public static VisualResults generate(RemoteWebDriver driver) {
+        return new VisualResults(driver);
+    }
+
+    private VisualResults(RemoteWebDriver driver) {
+        results = (Map<String, Object>) driver.executeScript("/*@visual.end*/");
+
+        this.passed = (Boolean) results.get("passed");
+        this.message = (String) results.get("message");
+        this.status = (String) results.get("status"); // success, failure, error, timeout, cancelled
```

Comment:
does it make sense to have an enum instead?

Reply:
I usually only do enums for Constructor/Method parameters so users don't have to remember the "magic string" to get the code to work correctly. We could do it for the return value here, but the use case for 99% of people is seeing the result displayed in stdout at the end of their test. I can't see anyone asserting on this or using it in logic anywhere.